Core/Core/Database/CoreData/Models/DiscussionTopic.swift

```diff
@@ -40,7 +44,44 @@ public final class DiscussionTopic: NSManagedObject, WriteableModel {
         model.postedAt = item.posted_at
         model.lastReplyAt = item.last_reply_at
         model.discussionSubEntryCount = item.discussion_subentry_count
+        model.message = item.message
         model.published = item.published
+        model.attachments = Set(try item.attachments?.map { attachment in
+            return try File.save(attachment, in: context)
+        } ?? [])
+        model.authorAvatarURL = item.author.avatar_image_url
+        model.authorID = item.author.id
+        model.authorName = item.author.display_name
         return model
     }
+
+    public var html: String {
```

Comment:
Seems like a lot of stuff happening here.  Would an extension on image or an html file clean this up a bit?

Reply:
IMO, The bulk of it is the svg of the attachment icon and since it's all pretty specific to the discussion topic I think it makes sense here. The svg could maybe be refactored out.

Reply:
I originally wanted to get a base64 data uri from our one of our icon images, but recoloring & resizing it was a bigger pain than just using the svg.